compiler/tests/org/jetbrains/kotlin/integration/CompilerSmokeTest.java

```diff
@@ -201,4 +201,50 @@ public void testNoReflect() throws Exception {
                      runCompiler("noReflect.compile", "-include-runtime", "-no-reflect", "noReflect.kt", "-d", jar));
         run("noReflect.run", "-cp", jar, "noReflect.NoReflectKt");
     }
+
+    // related to KT-14772, destination is not a jar, and clashing with an existing file
+    public void testDestinationDirClashingWithExistingFile() throws Exception {
+        String outputDir = tmpdir.getAbsolutePath() + File.separator + "clashingFile";
+        File file = new File(outputDir);
+        file.createNewFile();
+        runCompiler("test.compile", "test.kt", "-d", outputDir);
+    }
+
+    // related to KT-18184, destination is not a jar, and permission denied
+    public void testDestinationDirNoPermission() throws Exception {
+        String outputDir = tmpdir.getAbsolutePath() + File.separator + "noPermissionDir";
+        File file = new File(outputDir, "Test.class");
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        // file.getParentFile().setReadOnly(); // won't work on Windows
```

Comment:
I guess we could wrap this into `if (SystemInfo.isWindows)` to make sure we test the original issue at least on Linux, but the current way is also OK